src/zarr/strings.py

```diff
@@ -0,0 +1,87 @@
+"""This module contains utilities for working with string arrays across
+different versions of Numpy.
+"""
+
+from typing import Any, cast
+from warnings import warn
+
+import numpy as np
+
+# STRING_DTYPE is the in-memory datatype that will be used for V3 string arrays
+# when reading data back from Zarr.
+# Any valid string-like datatype should be fine for *setting* data.
+
+STRING_DTYPE: np.dtypes.StringDType | np.dtypes.ObjectDType
```

Comment:
I'm stuck on a tricky typing issue. However, the `np.dtypes.ObjectDType` is only available in numpy 2.0.

Is there a way to have the typing somehow depend on whether numpy 2.0 is available in the environment?